### ISSUE TITLE
[xcode] Failed to package, Command PhaseScriptExecution failed with a nonzero exit code - Flutter

### ISSUE THREAD

**Issue by HellRyder43** [flutter/flutter#104726]
I have built a flutter app. I can run the app in xCode simulator without no problem. But when I try to archive it, it throws the PhaseScriptExecution error. I have tried flutter clean then reinstall the pods again. But to no avail.

Here is my flutter doctor result:

```
[✓] Flutter (Channel stable, 3.0.1, on macOS 12.4 21F79 darwin-x64, locale
    en-MY)
    • Flutter version 3.0.1 at /Users/admin/Developer/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision fb57da5f94 (6 days ago), 2022-05-19 15:50:29 -0700
    • Engine revision caaafc5604
    • Dart version 2.17.1
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version
    32.1.0-rc1)
    • Android SDK at /Users/admin/Library/Android/sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Users/admin/Downloads/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 12.4 21F79 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 84.0.4147.105

[✓] HTTP Host Availability
    • All required HTTP hosts are available
```

And this is the output of `flutter build ios`

```
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    Writing result bundle at path:
        /var/folders/hd/glkbj0nj6ll0c_cgglq_cx400000gq/T/flutter_tools.sxyetd/flutt
        er_ios_build_temp_dirhsbaOo/temporary_xcresult_bundle


    Failed to package /Users/admin/StudioProjects/mymaikapp.
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Planning
    note: Build preparation complete
    note: Building targets in dependency order

    Result bundle written to path:
        /var/folders/hd/glkbj0nj6ll0c_cgglq_cx400000gq/T/flutter_tools.sxyetd/flutt
        er_ios_build_temp_dirhsbaOo/temporary_xcresult_bundle


Encountered error while building for device.
```

If it would help, I ran the command `flutter run --verbose` and put the log here, if it would help tracing what is the error, but I couldn't find it.

[[flutter run verbose output](https://pastebin.com/yhdt8pL1)](https://pastebin.com/yhdt8pL1)

Any help would be greatly appreciated. Thank you for your time.

**Comment by haithngn** [comment]
I am facing with the same issue on a pure Flutter project.
```
flutter doctor -v
```

```
[✓] Flutter (Channel stable, 3.0.2, on macOS 11.6 20G165 darwin-x64, locale en-US)
    • Flutter version 3.0.2 at /Users/hai/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cd41fdd495 (7 days ago), 2022-06-08 09:52:13 -0700
    • Engine revision f15f824b57
    • Dart version 2.17.3
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/hai/Library/Android/sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 68.1.3
    • Dart plugin version 213.7371

[✓] VS Code (version 1.68.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (2 available)
    • iPhone X 14 (mobile) • 6EF4787A-ECF7-41A0-B7DD-B426ACABDB8F • ios        • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)      • macos                                • darwin-x64 • macOS 11.6 20G165 darwin-x64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
flutter run -v
```

```
[   +2 ms] Updating assets
[ +199 ms] Syncing files to device iPhone X 14...
[   +4 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[        ] Processing bundle.
[   +2 ms] <- recompile package:seed/main.dart cf8e15ac-6bf7-4538-860d-c98e6dd22be7
[        ] <- cf8e15ac-6bf7-4538-860d-c98e6dd22be7
[   +2 ms] Bundle processing done.
[ +166 ms] Updating files.
[        ] DevFS: Sync finished
[   +2 ms] Syncing files to device iPhone X 14... (completed in 178ms)
[   +1 ms] Synced 0.0MB.
[   +3 ms] <- accept
[  +13 ms] Connected to _flutterView/0x7ffbd381ec20.
[   +6 ms] Flutter run key commands.
[   +4 ms] r Hot reload. 🔥🔥🔥
[   +1 ms] R Hot restart.
[   +1 ms] h List all available interactive commands.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] 💪 Running with sound null safety 💪
[        ] An Observatory debugger and profiler on iPhone X 14 is available at: http://127.0.0.1:58479/97A3TMKRbnE=/
[ +279 ms] The Flutter DevTools debugger and profiler on iPhone X 14 is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:58479/97A3TMKRbnE=/
```

Build for real device succeed:
![Screen Shot 2022-06-15 at 2 30 10 PM](https://user-images.githubusercontent.com/22902671/173769078-1c15535d-5a2d-4a51-b25a-7aede32f9d1e.png)

The variable $FLUTTER_ROOT is defined correctly.
xcode_backend file path exists, the Run script locates that script correctly.

```
flutter build ios
```

```
Running Xcode build...
Xcode build done.                                           12.5s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/cc/zdwbbxd969bdq82dsltv3j480000gn/T/flutter_tools.fflLld/flutter_ios_build_temp_dirwyiZ0O/temporary_xcresult_bundle


    Failed to package /Users/hai/workspace/repositories/gitlab/seedart/seed.
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Planning
    note: Build preparation complete
    note: Building targets in dependency order

    Result bundle written to path:
    	/var/folders/cc/zdwbbxd969bdq82dsltv3j480000gn/T/flutter_tools.fflLld/flutter_ios_build_temp_dirwyiZ0O/temporary_xcresult_bundle
```

I hope you have a detail look into the weird issue.

**Comment by haithngn** [comment]
I solved this by downgrade flutter SDK to 3.0.1. 
I can deploy both iOS and Android without modifying the source code so I'm guessing the cause is inside xcode_backend.sh and xcode_backend.dart which had been changed from previous version 3.0.1

@maheshmnj could you please investigate them.

**Comment by thipokch** [comment]
@haithngn I'm having the same issues too. I had to downgrade from 3.0.4 to 3.0.1
